src/auto_archiver/modules/opentimestamps_enricher/opentimestamps_enricher.py

```diff
@@ -0,0 +1,184 @@
+                # Get the file path from the media
+                file_path = media.filename
+                if not os.path.exists(file_path):
+                    logger.warning(f"File not found: {file_path}")
+                    continue
+                
+                # Create timestamp for the file - hash is SHA256
+                # Note: ONLY SHA256 is used/supported here. Opentimestamps supports other hashes, but not SHA3-512
+                # see opentimestamps.core.op
+                logger.debug(f"Creating timestamp for {file_path}")
+                file_hash = None
+                with open(file_path, 'rb') as f:
+                    file_hash = OpSHA256().hash_fd(f)
+
+                if not file_hash:
+                    logger.warning(f"Failed to hash file for timestamping, skipping: {file_path}")
+                    continue
+                
+                # Create a timestamp with the file hash
+                timestamp = Timestamp(file_hash)
+                
+                # Create a detached timestamp file with the hash operation and timestamp
+                detached_timestamp = DetachedTimestampFile(OpSHA256(), timestamp)
+                
+                # Submit to calendar servers
+                submitted_to_calendar = False
+                if self.use_calendars:
+                    logger.debug(f"Submitting timestamp to calendar servers for {file_path}")
+                    calendars = []
+                    whitelist = DEFAULT_CALENDAR_WHITELIST
+                    
+                    if self.calendar_whitelist:
+                        whitelist = set(self.calendar_whitelist)
+                    
+                    # Create calendar instances
+                    calendar_urls = []
+                    for url in self.calendar_urls:
+                        if url in whitelist:
+                            calendars.append(RemoteCalendar(url))
+                            calendar_urls.append(url)
+                    
+                    # Submit the hash to each calendar
+                    for calendar in calendars:
+                        try:
+                            calendar_timestamp = calendar.submit(file_hash)
+                            timestamp.merge(calendar_timestamp)
+                            logger.debug(f"Successfully submitted to calendar: {calendar.url}")
+                            submitted_to_calendar = True
+                        except Exception as e:
+                            logger.warning(f"Failed to submit to calendar {calendar.url}: {e}")
+                    
+                    # If all calendar submissions failed, add pending attestations
+                    if not submitted_to_calendar and not timestamp.attestations:
+                        logger.info("All calendar submissions failed, creating pending attestations")
+                        for url in calendar_urls:
+                            pending = PendingAttestation(url)
```

Comment:
I may be misunderstanding what you are trying to do here. But I think this is incorrect and could cause a lot of confusion. `PendingAttestations` are for when a calendar has returned a commitment to make a timestamp in the future; they're not meant to be used as placeholders for a timestamp that failed due to unreachable calendars.

Reply:
You're right, we shouldn't be doing this. I'll refactor. Thanks again for the code review!

Reply:
Glad to help!

I'll let you know if the people running that IPv4-only calendar add IPv6.